### PR TITLE
Move fuse conv bn to mmcv

### PR DESCRIPTION
**Pull request by yhcao6**

### CODE REVIEW

File: mmcv/cnn/utils/fuse_conv_bn.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        bn.running_mean)
+
+    factor = bn.weight / torch.sqrt(bn.running_var + bn.eps)
+    conv.weight = nn.Parameter(conv_w *
+                               factor.reshape([conv.out_channels, 1, 1, 1]))
+    conv.bias = nn.Parameter((conv_b - bn.running_mean) * factor + bn.bias)
+    return conv
+
+
+def fuse_module(m):
```
Comment (hellock) [inline mmcv/cnn/utils/fuse_conv_bn.py:22]: Docstring

File: tests/test_cnn/test_fuse_conv_bn.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import torch
+
+from mmcv.cnn import ConvModule, fuse_module
+
+
+def test_fuse_module():
+    inputs = torch.rand((1, 3, 5, 5))
+    module = ConvModule(3, 5, 3, norm_cfg=dict(type='BN'))
+    fused_module = fuse_module(module)
+    assert torch.equal(module(inputs), fused_module(inputs))
```
Comment (hellock) [inline tests/test_cnn/test_fuse_conv_bn.py:10]: Improve the coverage.

File: mmcv/cnn/utils/fuse_conv_bn.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import torch
+import torch.nn as nn
+
+
+def fuse_conv_bn(conv, bn):
+    """ During inference, the functionary of batch norm layers is turned off
+    but only the mean and var alone channels are used, which exposes the
+    chance to fuse it with the preceding conv layers to save computations and
+    simplify network structures.
```
Comment (hellock) [inline mmcv/cnn/utils/fuse_conv_bn.py:9]: Docstring

File: mmcv/cnn/utils/fuse_conv_bn.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    conv.bias = nn.Parameter((conv_b - bn.running_mean) * factor + bn.bias)
+    return conv
+
+
+def fuse_module(m):
+    last_conv = None
+    last_conv_name = None
+
+    for name, child in m.named_children():
+        if isinstance(child, (nn.BatchNorm2d, nn.SyncBatchNorm)):
```
Comment (hellock) [inline mmcv/cnn/utils/fuse_conv_bn.py:27]: nn.BatchNorm1d?

File: mmcv/cnn/utils/fuse_conv_bn.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        bn.running_mean)
+
+    factor = bn.weight / torch.sqrt(bn.running_var + bn.eps)
+    conv.weight = nn.Parameter(conv_w *
+                               factor.reshape([conv.out_channels, 1, 1, 1]))
+    conv.bias = nn.Parameter((conv_b - bn.running_mean) * factor + bn.bias)
+    return conv
+
+
+def fuse_conv_bn(m):
```
Comment (hellock) [inline mmcv/cnn/utils/fuse_conv_bn.py:31]: m -> module

File: mmcv/cnn/utils/fuse_conv_bn.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    """Recursively fuse conv and bn in a module.
+
+    Args:
+        m: Module to be fused.
```
Comment (hellock) [inline mmcv/cnn/utils/fuse_conv_bn.py:35]: m (nn.Module): xxx

File: mmcv/cnn/utils/fuse_conv_bn.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+import torch.nn as nn
+
+
+def fuse_conv_bn_single(conv, bn):
+    """Fuse conv and bn into one module.
+
+    During inference, the functionary of batch norm layers is turned off
+    but only the mean and var alone channels are used, which exposes the
+    chance to fuse it with the preceding conv layers to save computations and
+    simplify network structures.
```
Comment (hellock) [inline mmcv/cnn/utils/fuse_conv_bn.py:11]: Move these doc to the public api `fuse_conv_bn()`.

File: mmcv/cnn/utils/fuse_conv_bn.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
+import torch
+import torch.nn as nn
+
+
+def fuse_conv_bn_single(conv, bn):
```
Comment (hellock) [inline mmcv/cnn/utils/fuse_conv_bn.py:5]: _fuse_conv_bn()

File: tests/test_cnn/test_fuse_conv_bn.py [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import torch
+import torch.nn as nn
+
+from mmcv.cnn import ConvModule, fuse_conv_bn
+
+
+def test_fuse_module():
```
Comment (hellock) [inline tests/test_cnn/test_fuse_conv_bn.py:7]: test_fuse_conv_bn

**Comment by hellock** [comment]
There are conflicts now.
